components/tidb_query_vec_executors/src/index_scan_executor.rs

```diff
@@ -344,148 +342,163 @@ impl IndexScanExecutorImpl {
-            DecodeIntHandle if key_payload.is_empty() => {
-                // This is a unique index, and we should look up PK int handle in the value.
-                let handle_val = self.decode_handle_from_value(value)?;
+            // This is a non-unique index value, we should extract the int handle from the key.
+            DecodeIntHandle if tail_len < 8 => {
+                datum::skip_n(&mut key_payload, self.columns_id_without_handle.len())?;
+                let handle = self.decode_handle_from_key(key_payload)?;
                 columns[self.columns_id_without_handle.len()]
                     .mut_decoded()
-                    .push_int(Some(handle_val));
+                    .push_int(Some(handle));
             }
+            // This is a unique index value, we should extract the int handle from the value.
             DecodeIntHandle => {
-                // This is a normal index, and we should look up PK handle in the key.
-                let handle_val = self.decode_handle_from_key(key_payload)?;
+                let handle = self.decode_handle_from_value(tail)?;
                 columns[self.columns_id_without_handle.len()]
                     .mut_decoded()
-                    .push_int(Some(handle_val));
+                    .push_int(Some(handle));
             }
+
             DecodeCommonHandle => {
-                // Otherwise, if the handle is common handle, we extract it from the key.
-                Self::extract_columns_from_datum_format(
-                    &mut key_payload,
-                    &mut columns[self.columns_id_without_handle.len()..],
-                )?;
+                if common_handle_bytes.is_empty() {
+                    // This is a non-unique index value, we should extract the int handle from the key.
+                    datum::skip_n(&mut key_payload, self.columns_id_without_handle.len())?;
+                    Self::extract_columns_from_datum_format(
+                        &mut key_payload,
+                        &mut columns[self.columns_id_without_handle.len()..],
+                    )?;
+                } else {
+                    // This is a unique index value, we should extract the int handle from the value.
+                    let mut handle = common_handle_bytes;
+                    let end_index = if self.decode_partition_id {
+                        columns.columns_len() - 1
+                    } else {
+                        columns.columns_len()
+                    };
+                    Self::extract_columns_from_datum_format(
+                        &mut handle,
+                        &mut columns[self.columns_id_without_handle.len()..end_index],
+                    )?;
+                }
             }
         }
 
+        if self.decode_partition_id {
+            // If need partition id, append partition id to the last column.
+            let pid = NumberCodec::decode_i64(partition_id_bytes);
```

Comment:
`partition_id_bytes.read_i64()`?

Reply:
Here bytes are encoded to comparable format in tidb, so must use `decode_i64` to decode, which first read `u64` and then convert to `i64`. See `components/codec/src/number.rs:150`.